data/lua/wml/harm_unit.lua

```diff
@@ -160,6 +160,32 @@ function wml_actions.harm_unit(cfg)
 
 			wesnoth.interface.float_label( unit_to_harm.x, unit_to_harm.y, string.format( "<span foreground='red'>%s</span>", text ) )
 
+			local xp_mode = {kill=false, attack=false, defend=false}
+			local experience_split = tostring(experience):split()
+			for _,opt in ipairs(experience_split) do
+				if opt == 'true' or opt == 'yes' or opt == 'nil' then
+					xp_mode.kill = true
+					xp_mode.attack = true
+					xp_mode.defend = true
+				elseif opt == 'fight' then
+					xp_mode.attack = true
+					xp_mode.defend = true
+				elseif opt == 'attack' then
+					xp_mode.attack = true
+				elseif opt == 'defend' then
+					xp_mode.defend = true
+				elseif opt == 'kill' then
+					xp_mode.kill = true
+				elseif opt == 'no' or opt == 'false' then
+					-- preserve previous value
```

Comment:
@CelticMinstrel what is the most proper way to get rid of luacheck of this branch? I guess it wants `elseif not (opt == 'no' or opt == 'false')`.

Reply:
I guess you'd either apply [DeMorgan's Law](https://en.wikipedia.org/wiki/De_Morgan%27s_laws), or use an [inline comment](https://luacheck.readthedocs.io/en/stable/inline.html).

Reply:
I saw the inline comment page, but did not understand from it which comment would work.
`elseif opt == 'no' or opt == 'false' then -- luacheck: ignore` looks like way to ignore one line.